compiler/noirc_frontend/src/hir/def_map/mod.rs

```diff
@@ -298,6 +305,23 @@ impl CrateDefMap {
             String::new()
         }
     }
+
+    fn check_unused_imports(
+        context: &mut Context,
+        crate_id: CrateId,
+        errors: &mut Vec<(CompilationError, FileId)>,
+    ) {
+        for module in context.def_maps[&crate_id].modules() {
+            for ident in module.unused_imports() {
+                errors.push((
+                    CompilationError::ResolverError(ResolverError::UnusedImport {
+                        ident: ident.clone(),
+                    }),
+                    module.location.file,
+                ));
+            }
```

Comment:
Can replace this with a `.map` and a `.extend` to avoid repeated pushing.

Reply:
Done!